evennia/typeclasses/attributes.py

```diff
@@ -414,6 +470,366 @@ def reset_cache(self):
         self._cache = {}
         self._catcache = {}
 
+    def create_attr(self, keystr, category, lockstring, value):
+        """
+        Creates Attribute (using the class specified for the backend) and returns it.
+        
+        This MUST actively save the Attribute to whatever database backend is used, AND
+        call self.set_cache(key, category, new_attrobj)
+        
+        Args:
+            keystr (str): The Attribute's key.
+            category (str or None): The Attribute's category, or None
+            lockstring (str): Any locks for the Attribute.
+            value (obj): The Value of the Attribute.
+
+        Returns:
+            attr (IAttribute): The new Attribuite.
+        """
+        raise NotImplementedError()
+
+    def update_attr(self, attr, value):
+        """
+        Simply sets a new Value to an Attribute.
+
+        Args:
+            attr (IAttribute): The Attribute being changed.
+            value (obj): The Value for the Attribute.
+
+        """
+        raise NotImplementedError()
+
+    def batch_add_update(self, attr_obj, category, lock_storage, new_value):
```

Comment:
For naming consistency, this would be `batch_add_update_attributes`. Since it's treated as an update-operation by the back-end, then maybe shortening to `batch_update_attributes` would be enough (we don't care here that we call it from a method named `batch_add` after all).